FirebaseSegmentation/Sources/SEGContentManager.m

```diff
@@ -94,37 +93,51 @@ - (void)associateCustomInstallationIdentiferNamed:(NSString *)customInstallation
       return;
     }
 
-    strongSelf->_instanceIdentifier = result.instanceID;
-    strongSelf->_instanceIdentifierToken = result.token;
-
-    NSMutableDictionary<NSString *, NSString *> *appAssociationData =
-        [[NSMutableDictionary alloc] init];
-    [appAssociationData setObject:customInstallationID forKey:kSEGCustomInstallationIdentifierKey];
-    [appAssociationData setObject:self->_instanceIdentifier
-                           forKey:kSEGFirebaseInstallationIdentifierKey];
-    [appAssociationData setObject:kSEGAssociationStatusPending forKey:kSEGAssociationStatusKey];
-    [strongSelf->_associationData setObject:appAssociationData forKey:firebaseApp];
-
-    // Update the database async.
-    // TODO(mandard) The database write and corresponding completion handler needs to be wired up
-    // once we support listening to FID changes.
-    [strongSelf->_databaseManager insertMainTableApplicationNamed:firebaseApp
-                                         customInstanceIdentifier:customInstallationID
-                                       firebaseInstanceIdentifier:strongSelf->_instanceIdentifier
-                                                associationStatus:kSEGAssociationStatusPending
-                                                completionHandler:nil];
-
-    // Send the change up to the backend. Also add the token.
-
-    [strongSelf->_networkManager
-        makeAssociationRequestToBackendWithData:appAssociationData
-                                          token:strongSelf->_instanceIdentifierToken
-                                     completion:^(BOOL status,
-                                                  NSDictionary<NSString *, id> *result) {
-                                       // TODO...log, update database.
-
-                                       completionHandler(status, result);
-                                     }];
+    strongSelf->_installationIdentifier = identifier;
+
+    [[self installationForApp:firebaseApp] authTokenWithCompletion:^(
+                                          FIRInstallationsAuthTokenResult *__nullable tokenResult,
+                                          NSError *_Nullable error) {
+      if (!tokenResult || error) {
+        NSString *errorMessage = @"Error getting AuthToken.";
+        if (error) {
+          errorMessage = [errorMessage stringByAppendingString:error.description];
+        }
+        NSDictionary *errorDictionary = @{kErrorDescription : errorMessage};
+        completionHandler(NO, errorDictionary);
+        return;
+      }
+      strongSelf->_installationIdentifierToken = tokenResult.authToken;
```

Comment:
If `strongSelf` is used here then `self` will be captured by this block. Lines 80-84 should be copied to the beginning of the block.